substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/HeapPolicyOptions.java

```diff
@@ -74,4 +74,13 @@
 
     @Option(help = "Defines the upper bound for the number of remaining bytes in the young generation that cause a collection when `System.gc` is called.") //
     public static final RuntimeOptionKey<Long> UserRequestedGCThreshold = new RuntimeOptionKey<>(16L * 1024L * 1024L);
+
+    @Option(help = "Maximum value for survivor space") //
+    public static final HostedOptionKey<Integer> MaxSurvivors = new HostedOptionKey<>(15);
+
+    @Option(help = "Number of survivor spaces") //
+    public static final RuntimeOptionKey<Integer> NumberOfSurvivors = new RuntimeOptionKey<>(4);
```

Comment:
Yes, we make it a `RuntimeOptionKey` for not build the image again(which could be slow at sometime), the garbage collections can happen before this value gets parsed? We just follow the the keys like `MaximumYoungGenerationSizePercent`.
I'm not sure if this can create issues here. Is there a better way to change the value as the options of image runtime?
If this can create issues, I can change the related options as the hosted option

Reply:
With the code as it is now, `NumberOfSurvivorSpaces` will have to be a `HostedOptionKey`.